atomic_reactor/plugins/exit_koji_promote.py

```diff
@@ -326,43 +332,93 @@ def get_buildroot(self, build_id):
 
         return buildroot
 
+    def parse_orchestrator_logs(self, logs):
+        """
+        Parse the output of get_orchestrator_build_logs()
+
+        :return: list, Output instances
+        """
+        output = []
+
+        # Build list of arches
+        arches = []
+        for log in logs:
+            if log.platform is not None and log.platform not in arches:
+                arches.append(log.platform)
+        for arch in arches:
+            # Delete once closed
+            logfile = NamedTemporaryFile(prefix="%s-%s" % (self.build_id, arch),
+                                         suffix=".log", mode='wb')
+            arch_logs = (log.line for log in logs if log.platform == arch)
+            logfile.write(arch_logs)
+            logfile.flush()
+            metadata = self.get_output_metadata(logfile.name, "%s.log" % arch)
+            output.append(Output(file=logfile, metadata=metadata))
+
+        logfile = NamedTemporaryFile(prefix="%s-orchestrator" % self.build_id,
+                                     suffix=".log", mode='wb')
+        orchestrator_logs = (log.line for log in logs if log.platform is None)
+        logfile.write(orchestrator_logs)
+        logfile.flush()
+        metadata = self.get_output_metadata(logfile.name, "orchestrator.log")
+        output.append(Output(file=logfile, metadata=metadata))
+
+        return output
+
+    def parse_logs_compat(self, logs):
+        """
+        Parse the output of get_build_logs()
+
+        :return: list, Output instances
+        """
+        output = []
+
+        # Deleted once closed
+        logfile = NamedTemporaryFile(prefix=self.build_id, suffix=".log", mode='wb')
+        try:
+            logfile.write(logs)
+        except (TypeError, UnicodeEncodeError):
+            # Older osbs-client versions returned Unicode objects
+            logfile.write(logs.encode('utf-8'))
```

Comment:
Correct, this was just some pre-existing code that got moved around a little bit.